soc/nordic/nrf53/nrf53_cpunet_mgmt.c

```diff
@@ -15,14 +15,40 @@
 #include <zephyr/sys/__assert.h>
 #include <zephyr/sys/notify.h>
 #include <zephyr/sys/onoff.h>
+#include <soc_secure.h>
 
 #include <hal/nrf_reset.h>
 #include <hal/nrf_spu.h>
 
 static struct onoff_manager cpunet_mgr;
 
+#define GPIOS_PSEL_BY_IDX(node_id, prop, idx) \
+	NRF_DT_GPIOS_TO_PSEL_BY_IDX(node_id, prop, idx),
+#define ALL_GPIOS_IN_NODE(node_id) \
+	DT_FOREACH_PROP_ELEM(node_id, gpios, GPIOS_PSEL_BY_IDX)
+#define ALL_GPIOS_IN_FORWARDER(node_id) \
+	DT_FOREACH_CHILD(node_id, ALL_GPIOS_IN_NODE)
+
+static void assign_pins(nrf_gpio_pin_sel_t sel)
+{
+#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
```

Comment:
Now this is incorrect.

I suggest `ifdef CONFIG_SOC_NRF_GPIO_FORWARDER_FOR_NRF5340` as it was, instead of relying on other configs like it was.

SOC_NRF_GPIO_FORWARDER_FOR_NRF5340 depends on NRF_SOC_SECURE_SUPPORTED,
NRF_SOC_SECURE_SUPPORTED depends on  !TRUSTED_EXECUTION_NONSECURE || (BUILD_WITH_TFM && TFM_PARTITION_PLATFORM).

Don't duplicate this logic, you can do the forwarding if TF-M is enabled with the platform partition enabled.